### ISSUE TITLE
Recordings download option

### ISSUE THREAD

**Issue by Eyerex** [dave-p/TVHadmin-JS#7]
Hello just wondered if it's possible to add a recordings download option to finished recordings ?

**Comment by dave-p** [comment]
At present TVHadmin uses the Tvheadend '/play' API if you try to play an existing recording. This downloads a .m3u8 playlist file which contains the URL for the recording. On my default browser (Firefox) VLC is set as the default application to open .m3u8 files and so the recording immediately starts playing in VLC. Chromium-based browsers however just download the .m3u8 file, it takes another click on the just-downloaded filename in the browser to start playback.

I had hoped that it would be possible to change the TVHadmin playback button to use the Tvheadend '/dvrfile' API, which supplies the .ts file directly. This would have simplified things for Chromium-based users as well as giving you a way to directly download the file. However when I tested the change, all browsers downloaded the entire recording before starting playback - not a good idea!

I'm not sure it's a good idea to add an extra button to the Recordings screen, especially as the behaviour depends on which browser is being used. If you have only an occasional requirement to download files, you can copy the URL from the Playback button (for example http://IP:9981/play/dvrfile/0382ccf027ef34b62759f2140dd1d975?title=New:%20Britain%27s%20Beautiful%20Rivers...), delete the '/play' and the title (eg http://IP:9981/dvrfile/0382ccf027ef34b62759f2140dd1d975) and enter that URL into your browser.

I will carry on looking to see if there is a better way to handle playback which works on all browsers.

**Comment by Eyerex** [comment]
No worries as have a samba server setup to access recordings just though it would be nice to have access via the recording menu

**Comment by stpf99** [comment]
hi , i'm searching for how to add play button "on current page" in TVHadmin", im complet newbie in web programming (js ...) so far in my test : 
-simply html5 video tag can play on Chrome tvh extracted from m3u link 
- i found https://www.draketo.de/software/m3u-player.js but no tested
-  option for extract m3u > https://github.com/aitorciki/jquery-playlist for embed i video-tag
- also if it is possible better way for proper tvh link is use API and commit new programs after click to player video area  new tab
